src/dsl/static.jl

```diff
@@ -195,10 +196,18 @@ end
 
 # return foo (must be a symbol)
 function parse_return!(stmts::Vector{Expr}, bindings, line::Expr)
-    if line.head != :return || !isa(line.args[1], Symbol)
+    if line.head != :return
+        return false
+    end
+    if isa(line.args[1], Expr) && line.args[1].head == :macrocall
+        var = gensym()
+        typ = QuoteNode(Any)
+        res = parse_trace_expr!(stmts, bindings, var, line.args[1], typ)
```

Comment:
```suggestion
        if !parse_trace_expr!(stmts, bindings, var, line.args[1], typ)
            return false # the right-hand-side is a macro but not a valid `@trace` expression
        end
```

Reply:
Thanks for the correction! I didn't realize that I was eating the false return on failure here.